### PR TITLE
feat: add support for serializing enums

### LINKED ISSUE #19: Enum and other types
**Is your feature request related to a problem? Please describe.**
I'd like to be able to parse `enum`s (for example [S3 Object#ChecksumAlgorithm](https://docs.aws.amazon.com/AmazonS3/latest/API/API_Object.html) is a clear `enum`).
Also it would be nice if it would be possible to parse arbitrary type if it has `factory` constructor `parse(String string)`/`static` function `tryParse(String string)` or `factory` constructor `fromXmlElement(XmlElement element)`.

**Describe the solution you'd like**
Test if required type has has `factory` constructor `parse(String input)` or `factory` constructor `fromXmlElement(XmlElement element)` and use it.

**Describe alternatives you've considered**
Don't have any yet.

**Additional context**
`Enum` from link could be like this:
```dart
enum ChecksumAlgorithm {
  crc32('CRC32'),
  crc32c('CRC32C'),
  sha1('SHA1'),
  sha256('SHA256');

  const ChecksumAlgorithm(this.value);

  factory ChecksumAlgorithm.fromXmlElement(XmlElement element) =>
    values.firstWhere((_enum) => _enum.value == element.text);

  final String value;
}
```

### PR DESCRIPTION
**Pull request by marius-h**
**Describe the change**
Added support for enums similar to how `json_serializable` works.
I also already added unit tests to cover the new feature and updated the example and readme to show how it works.

Testing `@XmlValue()` for enum values was incredibly difficult and I'm still not fully satisfied with the code I wrote in terms of cleanliness but hey it works 🤷🏻‍♂️ 

Closes #19

### CODE REVIEW

**Comment by tnc1997** [comment]
Hello @marius-h, thank you very much for your contribution! I'll start reviewing as soon as I have a bit of time spare 😃

**Comment by marius-h** [comment]
@tnc1997 Awesome! Thank you! 😃

**Comment by tnc1997** [comment]
Could you remove the automatically generated `.dart.vm.json` files and revert any `.gitignore` changes if possible as it's automatically generated by https://www.toptal.com/developers/gitignore?templates=dart,intellij,visualstudiocode?

**Comment by marius-h** [comment]
Sure :)

File: xml_serializable/lib/src/generator_factories/serializer_generator_factory.dart [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import 'package:analyzer/dart/element/type.dart';
+import 'package:xml_serializable/src/serializer_generators/enum_serializer_generator.dart';
```
Comment (tnc1997) [inline xml_serializable/lib/src/generator_factories/serializer_generator_factory.dart:2]: Could we make this path relative to keep it consistent with the others in this file and the rest of the package?

```suggestion
import '../serializer_generators/enum_serializer_generator.dart';
```

**Review by tnc1997 — COMMENTED** [review]
Thank you very much again for your contribution, it is very much appreciated! I have reviewed everything except for the tests, as these might change based on some of the discussions.

**Comment by tnc1997** [comment]
Thank you very much again for your contribution, I am aiming to release these changes within the next week.

**Pull request merged**